app/recurring_job.go

```diff
@@ -506,6 +485,41 @@ func (job *Job) deleteSnapshots(names []string, volume *longhornclient.Volume, v
 	return nil
 }
 
+func (job *Job) purgeSnapshots(volume *longhornclient.Volume, volumeAPI longhornclient.VolumeOperations) error {
+	if _, err := volumeAPI.ActionSnapshotPurge(volume); err != nil {
+		return err
+	}
+	for {
+		done := true
```

Comment:
Just realized that this is an infinite loop... Should we add a timeout mechanism?

Reply:
Done!